File: web/admin/src/api/jointrun.ts
import client from './client'

export interface JointRun {
  id: number
  club_id: number
  date: string
  location_id: number
  creator_id: number
  created_at?: string
  updated_at?: string
}

export interface JointRunCreate {
  date: string
  location_id: number
  creator_id: number
}

export interface JointRunParticipant {
  id: number
  joint_run_id: number
  member_id: number
  member_fio?: string
  created_at?: string
}

export default {
  list(clubId: number) {
    return client.get<JointRun[]>(`/clubs/${clubId}/joint-runs`)
  },

  create(clubId: number, data: JointRunCreate) {
    return client.post<JointRun>(`/clubs/${clubId}/joint-runs`, data)
  },

  get(id: number) {
    return client.get<JointRun>(`/joint-runs/${id}`)
  },

  remove(id: number) {
    return client.delete(`/joint-runs/${id}`)
  },

  participants(runId: number) {
    return client.get<JointRunParticipant[]>(`/joint-runs/${runId}/participants`)
  },
}

File: internal/domain/entity/jointrun.go
package entity

import "time"

type JointRun struct {
	ID         int64     `json:"id"          db:"id"`
	ClubID     int64     `json:"club_id"     db:"club_id"`
	LocationID int64     `json:"location_id" db:"location_id"`
	CreatorID  int64     `json:"creator_id"  db:"creator_id"`
	Date       time.Time `json:"date"        db:"date"`
	MessageID  int64     `json:"message_id"  db:"message_id"`
	CreatedAt  time.Time `json:"created_at"  db:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"  db:"updated_at"`
}

type JointRunParticipant struct {
	ID         int64 `json:"id"           db:"id"`
	JointRunID int64 `json:"joint_run_id" db:"joint_run_id"`
	MemberID   int64 `json:"member_id"    db:"member_id"`
}
